Assets/Scripts/Models/Area/Temperature.cs

```diff
@@ -258,17 +258,17 @@ private int GetIndex(int x, int y, int z)
     /// </summary>
     private void ProgressTemperature(float deltaT) 
     {
+        Thread thread = new Thread(ForwardTemp);
+        thread.Start();
+
         // TODO: Compute temperature sources.
         if (sinksAndSources != null) 
         {
             foreach (Action<float> act in sinksAndSources.Values) 
             {
                 act(deltaT);
             }
-        }
-
-        Thread thread = new Thread(ForwardTemp);
-        thread.Start();
+        }     
```

Comment:
No idea honestly... I just stumbled on the first two. I can fix it real quick with a script though, if you don't mind.